stairway/src/main/java/bio/terra/stairway/DefaultThreadPoolTaskExecutor.java

```diff
@@ -0,0 +1,31 @@
+package bio.terra.stairway;
+
+import org.springframework.scheduling.concurrent.ThreadPoolTaskExecutor;
+
+public class DefaultThreadPoolTaskExecutor extends ThreadPoolTaskExecutor {
+
+  static final int DEFAULT_MAX_PARALLEL_FLIGHTS = 20;
+
+  /**
+   * An initialized {@link ThreadPoolTaskExecutor}.
+   *
+   * @param maxParallelFlights the desired pool size, honored if a positive integer, defaults to
+   *     DEFAULT_MAX_PARALLEL_FLIGHTS otherwise.
+   */
+  public DefaultThreadPoolTaskExecutor(Integer maxParallelFlights) {
+    super();
+    int poolSize = getPoolSize(maxParallelFlights);
+    super.setCorePoolSize(poolSize);
+    super.setMaxPoolSize(poolSize);
+    super.setKeepAliveSeconds(0);
+    super.setThreadNamePrefix("stairway-thread-");
+    super.initialize();
+  }
+
+  private int getPoolSize(Integer maxParallelFlights) {
+    if (maxParallelFlights == null || maxParallelFlights <= 0) {
```

Comment:
Just because it looks like you're switching using the syntax in StairwayImpl, this should probably be:
```suggestion
    if (Optional.ofNullable(maxParallelFlights).orElse(0) <= 0) {
```

But it's a whole character longer :)

Reply:
I'm actually going to walk back that syntax change, so I'll leave this line as it stands, thanks!

Reply:
Absolutely.  Makes sense!